RocketChat/Rocket.Chat.ReactNative
# chore: migrate WorkspaceView to hooks

## Proposed changes

migrate WorkspaceView to hooks

## Issue(s)	


## How to test or reproduce

The WorkspaceView is the view where appears the Login button and Create an Account

- Test with iframe's login server
- Test with servers that don't accepts more account to create

## Screenshots
![Screen Shot 2023-04-18 at 13 04 22](https://user-images.githubusercontent.com/47038980/232836557-5899aeba-46d3-46b2-8bb4-1d7fe603265b.png)

## Types of changes




## Checklist



## Further comments

[NATIVE-6]

[NATIVE-6]: https://rocketchat.atlassian.net/browse/NATIVE-6?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
